src/frequenz/sdk/config/_manager.py

```diff
@@ -0,0 +1,99 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Management of configuration."""
+
+import pathlib
+from collections.abc import Mapping, Sequence
+from datetime import timedelta
+from typing import Any, Final
+
+from frequenz.channels import Broadcast, Receiver
+
+from ._managing_actor import ConfigManagingActor
+
+
+class ConfigManager:
+    """A manager for configuration files.
+
+    This class reads configuration files and sends the configuration to the receivers,
+    providing optional configuration key filtering and schema validation.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        config_paths: Sequence[pathlib.Path],
+        /,
+        *,
+        auto_start: bool = True,
+        force_polling: bool = True,
+        name: str | None = None,
+        polling_interval: timedelta = timedelta(seconds=5),
+    ) -> None:
+        """Initialize this config manager.
+
+        Args:
+            config_paths: The paths to the TOML files with the configuration. Order
+                matters, as the configuration will be read and updated in the order
+                of the paths, so the last path will override the configuration set by
+                the previous paths. Dict keys will be merged recursively, but other
+                objects (like lists) will be replaced by the value in the last path.
+            auto_start: Whether to start the actor automatically. If `False`, the actor
+                will need to be started manually by calling `start()` on the actor.
+            force_polling: Whether to force file polling to check for changes.
+            name: A name to use when creating actors. If `None`, `str(id(self))` will
+                be used. This is used mostly for debugging purposes.
+            polling_interval: The interval to poll for changes. Only relevant if
+                polling is enabled.
+        """
+        self.name: Final[str] = str(id(self)) if name is None else name
```

Comment:
The idea is to rename `name` to `unique_id`, this will come when we finish the `core` library and we migrate to it. Once name is `unique_id`, it doesn't make sense to include the class name in it.

But the `ConfigManager` should really be a `BackgroundService`, I will probably do that, so the default `name`/`unique_id` will be done by `BackgroundService` as with actors.